File: sdk/dotnet/src/xjustiz.core-dotnet/Models/Entities/Core/NachrichtenkopfCore.cs
namespace xjustiz.core_dotnet.Models.Entities.Core;

using System.Xml.Serialization;
using xjustiz.core_dotnet.Models.Helpers;
using xjustiz.core_dotnet.Util.Versioning;

/// <summary>
/// Der Kopf der Nachricht für X.Justiz Core.<br/>
/// <u><b>Message header:</b></u> The header of the message for X.Justiz Core.
/// </summary>
[XJustizCoreAvailability(XJustizCoreVersion.V0_2_0)]
public class NachrichtenkopfCore : Nachrichtenkopf
{
    /// <summary>
    /// Verwendete X.Justiz Core Version für diese Nachricht. Redundant zu <see cref="Version_XJustizCore"/><br/>
    /// <u><b>XJustiz Core version:</b></u> Used X.Justiz Core version for this message. Redundant to <see cref="Version_XJustizCore"/>
    /// </summary>
    [XmlAttribute("xjustizVersion")]
    [XJustizAvailability(XJustizVersion.V2_1_0)]
    [XJustizCoreAvailability(XJustizCoreVersion.V0_2_0)]
    public new string Version { get; set; } = SchemeConstants.XJustizVersion.VersionString;

    /// <summary>
    /// Verwendete X.Justiz Core Version für diese Nachricht. Redundant zu <see cref="Version"/><br/>
    /// <u><b>XJustiz Core version:</b></u> Used X.Justiz Core version for this message. Redundant to <see cref="Version"/>
    /// </summary>
    [XmlElement("xjustizCoreVersion", Namespace = SchemeConstants.XJustiz_Tns, Order = 3)]
    [XJustizAvailability(XJustizVersion.V3_1_1)]
    [XJustizCoreAvailability(XJustizCoreVersion.V0_2_0)]
    public string Version_XJustizCore { get; set; } = SchemeConstants.XJustizVersion.VersionString;

    /// <summary>
    /// Verwendete X.Justiz Core Version für diese Nachricht als String.<br/>
    /// <u><b>XJustiz Core version:</b></u> Used X.Justiz Core version for this message as string.
    /// </summary>
    [XmlElement("xjustizReferenzVersion", Namespace = SchemeConstants.XJustiz_Tns, Order = 4)]
    [XJustizAvailability(XJustizVersion.V3_1_1)]
    [XJustizCoreAvailability(XJustizCoreVersion.V0_2_0)]
    public string Version_XJustizReferenz { get; set; } = SchemeConstants.XJustizVersion.XJustizReferenzVersionString;
}

File: sdk/java/src/main/java/de/xjustiz/core/models/NachrichtenkopfCore.java
package de.xjustiz.core.models;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import jakarta.xml.bind.annotation.XmlAccessType;
import jakarta.xml.bind.annotation.XmlAccessorType;
import jakarta.xml.bind.annotation.XmlElement;
import jakarta.xml.bind.annotation.XmlType;

/**
 * Der Kopf der Nachricht für X.Justiz Core.
 * <p>
 * <u><b>Message header:</b></u> The header of the message for X.Justiz Core.
 */
@XmlType(name = "NachrichtenkopfCore", propOrder = { "version_XJustizCore", "version_XJustizReferenz" })
@XmlAccessorType(XmlAccessType.FIELD)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class NachrichtenkopfCore extends Nachrichtenkopf {

    /**
     * Verwendete X.Justiz Core Version für diese Nachricht. Redundant zu {@link Version}
     * <p>
     * <u><b>XJustiz Core version:</b></u> Used X.Justiz Core version for this message. Redundant to {@link Version}
     */
    @XmlElement(name = "xjustizCoreVersion", namespace = "http://www.xjustiz.de")
    @JacksonXmlProperty(localName = "xjustizCoreVersion", namespace = "http://www.xjustiz.de")
    @JsonProperty("Version_XJustizCore")
    @JsonAlias({ "version_XJustizCore" })
    private String version_XJustizCore;

    /**
     * Verwendete X.Justiz Core Version für diese Nachricht als String.
     * <p>
     * <u><b>XJustiz Core version:</b></u> Used X.Justiz Core version for this message as string.
     */
    @XmlElement(name = "xjustizReferenzVersion", namespace = "http://www.xjustiz.de")
    @JacksonXmlProperty(localName = "xjustizReferenzVersion", namespace = "http://www.xjustiz.de")
    @JsonProperty("Version_XJustizReferenz")
    @JsonAlias({ "version_XJustizReferenz" })
    private String version_XJustizReferenz;

    /**
     * Default constructor.
     */
    public NachrichtenkopfCore() {}

    /**
     * Gets the value of the Version_XJustizCore property.
     * @return the value of the Version_XJustizCore property
     */
    public String getVersion_XJustizCore() { return version_XJustizCore; }
    /**
     * Sets the value of the Version_XJustizCore property.
     * @param version_XJustizCore the value to set
     */
    public void setVersion_XJustizCore(String version_XJustizCore) { this.version_XJustizCore = version_XJustizCore; }

    /**
     * Gets the value of the Version_XJustizReferenz property.
     * @return the value of the Version_XJustizReferenz property
     */
    public String getVersion_XJustizReferenz() { return version_XJustizReferenz; }
    /**
     * Sets the value of the Version_XJustizReferenz property.
     * @param version_XJustizReferenz the value to set
     */
    public void setVersion_XJustizReferenz(String version_XJustizReferenz) { this.version_XJustizReferenz = version_XJustizReferenz; }

}